Automattic/newspack-plugin
# feat(wc): duplicate orders admin notice

### All Submissions:






### Changes proposed in this Pull Request:



Adds an admin notice when duplicate orders are detected:

<img width="1165" alt="image" src="https://github.com/user-attachments/assets/49de1953-30b0-43d1-8a6e-9d76c91b9b14">

### How to test the changes in this Pull Request:

1. Donate two times with the same amount, within 10 minutes
2. Do that again, but change the dates of these two to more than a month ago
3. Trigger the CRON job: `wp cron event run newspack_wc_check_order_duplicates`
4. Visit the admin panel, observe the admin notice about potentially duplicated orders – from the last month (the first two created)
5. Dismiss one of the cases, observe the page reloads and the list item is no longer displayed
6. Run `wp newspack detect-order-duplicates --cutoff_time='1 year' --save` and refresh the admin page
7. Observe three items are present on the list: two new ones from more than a month ago

### Other information:




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208711277186607
